### PR TITLE
Adding support for onLayoutChange notifier configuration

### PR DESCRIPTION
**Pull request by sean-roberts**
We are trying to allow an integrator to know about the layout state of our sidebar as it changes.
The signature we are expecting is:

```
window.hypothesisConfig = ()=>{
  return {
    onLayoutChange: ( { width, height, expanded } )=>{}
  };
};
```

As noted above we will provide the width, height, and the expanded state.

### CODE REVIEW

**Comment by robertknight** [comment]
Rebased on master.

In testing this, I see that when opening or closing the sidebar the width reported is the one _before_ the state changed. This means that if an integrator tries to adjust the position of some elements when the sidebar becomes wider after it is opened or closed, it will end up in the wrong place:

Example output logged by an `onLayoutChanged` callback when opening and then closing the sidebar:

```
Sidebar state changed 37x316. Open: true
Sidebar state changed 465x316. Open: false
```

**Comment by robertknight** [comment]
Testing using localhost:3000, the callback is not invoked when the client initially loads if `openSidebar` is set to false.

**Steps to reproduce:**

1. Modify `live-reload-server.js`'s `hypothesisConfig` setup code to:
```js
window.hypothesisConfig = function () {
  return {
    liveReloadServer: 'ws://' + appHost + ':${port}',
    onLayoutChange({ width, height, expanded }) {
      console.log(\`Sidebar state changed \${width}x\${height}. Open: \${expanded}\`);
    }
  };
};
```
2. Restart gulp and browse to http://localhost:3000

**Expected:** Callback invoked when client initially loads with `{ width: <collapsed width>, height: <sidebar height>, expanded: false }`.
**Actual:** No log messages printed.

File: src/annotator/sidebar.coffee [hunk 1]
```diff
@@ -46,6 +46,8 @@ module.exports = class Sidebar extends Host
       @onProfileRequest = serviceConfig.onProfileRequest
       @onHelpRequest = serviceConfig.onHelpRequest
 
+    @onLayoutChange = config.onLayoutChange;
```
Comment (robertknight) [inline src/annotator/sidebar.coffee]: My [CoffeeScript syntax highlighter](https://github.com/kchmck/vim-coffee-script) complains if lines end with semicolons. The compiler doesn't complain but we should omit them for consistency.

File: src/annotator/sidebar.coffee [hunk 2]
```diff
@@ -120,6 +122,26 @@ module.exports = class Sidebar extends Host
         w = -m
         @frame.css('margin-left', "#{m}px")
         if w >= MIN_RESIZE then @frame.css('width', "#{w}px")
+        this._notifyOfLayoutChange()
+
+  _notifyOfLayoutChange: () =>
+    BUCKETBAR_WIDTH = 37
```
Comment (robertknight) [inline src/annotator/sidebar.coffee]: Would it be possible to avoid hardcoding this number?

File: src/annotator/config/index.js [hunk 3]
```diff
@@ -19,6 +19,7 @@ function configFrom(window_) {
 
     annotations: settings.annotations,
     branding: settings.hostPageSetting('branding'),
+    onLayoutChange: settings.hostPageSetting('onLayoutChange'),
```
Comment (robertknight) [inline src/annotator/config/index.js]: I wonder if we should rename this to `onSidebarLayoutChange` to be more explicit about what thing's layout changed?

Comment (sean-roberts) [inline src/annotator/config/index.js]: Sidebar is probably redundant. it's not sidebarBranding, etc. Let's keep it the way it is 👍

**Comment by sean-roberts** [comment]
Updated the width algorithm to calculate properly when we toggle expanded states. I updated the expanded state to also reflect if we are above the fully collapsed width - allowing panning to properly update expanded or not

**Pull request merged**